subworkflows/local/sourmash.nf

```diff
@@ -36,10 +38,25 @@ workflow SOURMASH {
                         save_prefetch,
                         save_prefetch_csv
                         )
+        ch_versions = ch_versions.mix(SOURMASH_GATHER.out.versions)
+
+        GUNZIP(SOURMASH_GATHER.out.result.map{ [ it[0], it[1] ] } )
+        SOURMASH_GATHER.out.result
+            .map{ it[1] }
+            .splitCsv( sep: ',', skip: 1 )
+            .map{ it[9] }
+            .set{ ch_accnos }
+
+        ch_accnos.view()
```

Comment:
Maybe it's because you read a _channel_ from `SOURMASH_GATHER`. The channel outputs _one_ entry (the csv file). Perhaps you do need to make a module, but can write a module in Nextflow alone that parses the file. Something like

```
input:
tuple val(meta), path(sourmash_csv)

output:
accessions

script:

splitCsv(sourmash_csv, header: true)
    .map { it.name }
    .set { accessions }
```

What I don't know, is how to specify the output channel for a module that doesn't have a normal script block (one with `"""`), but I think I saw a question regarding this in the Slack the other day.

Reply:
I can do this in the sub workflow- and I obtained the same results as above:
"GCA_002688505.1
"GCF_000008885.1 Wigglesworthia glossinidia endosymbiont of Glossina brevipalpis
"GCF_000455285.1 Candidatus Stoquefichus massiliensis AP9 strain=AP9
"GCF_000231445.1 Commensalibacter intestini A911 strain=A911"
etc...